openmdao/drivers/analysis_driver.py

```diff
@@ -0,0 +1,451 @@
+            as a dictionary whose keys are the variables to be recorded,
+            and whose associated values are dictionaries of metadata to
+            be passed on as keyword arguments to add_response.
+        """
+        if isinstance(responses, str):
+            self.add_response(responses)
+        elif isinstance(responses, dict):
+            for var, meta in responses.items():
+                self.add_response(var, **meta)
+        elif isinstance(responses, Iterable):
+            for res in responses:
+                if isinstance(res, str):
+                    self.add_response(res)
+
+    def _setup_comm(self, comm):
+        """
+        Perform any driver-specific setup of communicators for the model.
+
+        Parameters
+        ----------
+        comm : MPI.Comm or <FakeComm> or None
+            The communicator for the Problem.
+
+        Returns
+        -------
+        MPI.Comm or <FakeComm> or None
+            The communicator for the Problem model.
+        """
+        self._prev_sample_vars.clear()
+
+        self._problem_comm = comm
+
+        if not MPI:
+            return comm
+        else:
+            procs_per_model = self.options['procs_per_model']
+
+            full_size = comm.size
+            self._num_colors = size = full_size // procs_per_model
+            if full_size != size * procs_per_model:
+                raise RuntimeError("The total number of processors is not evenly divisible by the "
+                                   "specified number of processors per model.\n Provide a "
+                                   f"number of processors that is a multiple of {procs_per_model}, "
+                                   "or specify a number of processors per model that divides "
+                                   f"into {full_size}.")
+
+            color = self._color = comm.rank % size
+            new_comm = comm.Split(color)
+            return new_comm
+
+    def _get_name(self):
+        """
+        Get the name of this DOE driver and its case generator.
+
+        Returns
+        -------
+        str
+            The name of this DOE driver and its case generator.
+        """
+        return self._name
+
+    def run(self):
+        """
+        Generate samples and run the model for each set of generated input values.
+
+        Rank 0 will both manage the distribution of samples to the other procs and
+        serve as a worker running the samples.
+
+        All other procs just run samples.
+
+        Returns
+        -------
+        bool
+            Failure flag; True if failed to converge, False is successful.
+        """
+        comm = self._problem_comm
+        model = self._problem().model
+        self.result.reset()
+        self.iter_count = 0
+
+        # Variables allowed samples are the inputs or implicit outputs in the model.
+        # Non-model-inputs would just have their value overridden when evaluating the model.
```

Comment:
Non-model-inputs is somewhat confusing.  Aren't we really talking about inputs connected to IVCs?